frontend/app-development/features/dataModelling/hooks/useValidateSchemaName.test.ts

```diff
@@ -0,0 +1,120 @@
+}));
+
+jest.mock('app-shared/hooks/useStudioEnvironmentParams', () => ({
+  useStudioEnvironmentParams: jest.fn(),
+}));
+
+describe('useValidateSchemaName', () => {
+  beforeEach(() => {
+    (useStudioEnvironmentParams as jest.Mock).mockReturnValue({ org, app });
+    (useAppMetadataQuery as jest.Mock).mockReturnValue({ data: appMetaData });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should set nameError to empty string when name is valid', () => {
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    result.current.validateName('ValidName');
+
+    expect(result.current.nameError).toBe('');
+  });
+
+  it('should set error when name is empty', () => {
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    act(() => {
+      result.current.validateName('');
+    });
+
+    expect(result.current.nameError).toBe(textMock('validation_errors.required'));
+  });
+
+  it('should set error when name does not match regex', () => {
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    act(() => {
+      result.current.validateName('123InvalidName');
+    });
+
+    expect(result.current.nameError).toBe(textMock('schema_editor.error_invalid_datamodel_name'));
+  });
+
+  it('should set error when name exceeds max length', () => {
+    const longName = 'a'.repeat(SCHEMA_NAME_MAX_LENGTH + 1);
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    act(() => {
+      result.current.validateName(longName);
+    });
+
+    expect(result.current.nameError).toBe(
+      textMock('validation_errors.maxLength', { 0: SCHEMA_NAME_MAX_LENGTH }),
+    );
+  });
+
+  it('should set error when name exists in dataModels', () => {
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    act(() => {
+      result.current.validateName(dataModel1NameMock);
+    });
+
+    expect(result.current.nameError).toBe(
+      textMock('schema_editor.error_model_name_exists', {
+        newModelName: dataModel1NameMock,
+      }),
+    );
+  });
+
+  it('should set error when name exists in dataTypes', () => {
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    act(() => {
+      result.current.validateName('dataTypeId');
+    });
+
+    expect(result.current.nameError).toBe(textMock('schema_editor.error_data_type_name_exists'));
+  });
+
+  it('should set error when name is a C# reserved keyword', () => {
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    act(() => {
+      result.current.validateName('class');
+    });
+
+    expect(result.current.nameError).toBe(textMock('schema_editor.error_reserved_keyword'));
+  });
+
+  it('should handle norwegian characters in name', () => {
+    const { result } = renderHook(() => useValidateSchemaName(dataModels));
+
+    result.current.validateName('Valid_Name-æøåÆØÅ');
```

Comment:
looks like this name passes a lot more checks than handling æ,ø,å. Maybe the name `æøå` is a more concise way to test this case?

And maybe we should add other tests that covers more of what is valid with regards to the regEx? If it does not exist another place 🙈

Reply:
Thanks! 😄 

I tried disabling the frontend validation and created a model name `æææ`:
![bilde](https://github.com/user-attachments/assets/f14f5a6a-c4f8-4d2b-b317-538366f523f8)

Reply:
New issue: https://github.com/Altinn/altinn-studio/issues/14116